File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = INT_MIN , sum = 0;
        for(int i = 0 ; i < nums.size() ; i++){
            if(sum<0)   sum = 0;
            sum += nums[i];
            maxSum = max( maxSum , sum );
        }  
        return maxSum;  
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0 ;
        int maxSum = -99999 ;
        for(int num : nums){
            if(sum < 0){
                sum = 0 ;
            }
            sum += num ;
            maxSum = Math.max(maxSum , sum) ;
        }
        return maxSum ;
    }
}